src/sys/unix/epoll.rs

```diff
@@ -141,11 +141,6 @@ fn ioevent_to_epoll(interest: Ready, opts: PollOpt) -> u32 {
         kind |= EPOLLOUT;
     }
 
-    if UnixReady::from(interest).is_hup() {
```

Comment:
So this means passing HUP readiness is a no-op on epoll platforms right?

Reply:
No, `HUP` is an implicit interest w/ epoll (you always get it). The error here was requesting `RDHUP` which means something different.

Reply:
Right `EPOLLHUP` is implicit but it won't be possible to get `EPOLLRDHUP` anymore right?